internal/cluster/distribution/eks/eksadapter/node_pool_store.go

```diff
@@ -62,3 +62,25 @@ func (s nodePoolStore) CreateNodePool(
 
 	return nil
 }
+
+// ListNodePoolNames retrieves the node pool names for the cluster specified by
+// its cluster ID.
+func (s nodePoolStore) ListNodePoolNames(ctx context.Context, clusterID uint) (nodePoolNames []string, err error) {
+	var nodePoolModels []eksmodel.AmazonNodePoolsModel
+
+	err = s.db.Find(&nodePoolModels,
+		map[string]interface{}{
+			"cluster_id": clusterID,
+		},
+	).Error
+	if err != nil {
+		return nil, errors.WrapWithDetails(err, "fetching node pools from database failed", "clusterID", clusterID)
+	}
+
+	nodePoolNames = make([]string, len(nodePoolModels))
```

Comment:
Same thing about slice appending.

Reply:
Fixed in https://github.com/banzaicloud/pipeline/compare/2ad4cafb067c442a1cfbf1b9cd4071e298b269c8..f640984ef5dee7ae5770cc55c9ccc9741c0e760b.

Side-note: also fixed the 
```
database.pipeline.amazon_node_pools.cluster_id != database.pipeline.clusters.id database.pipeline.amazon_node_pools.cluster_id == database.pipeline.amazon_eks_clusters.id (in case of EKS clusters)
```
issue here at the node pool name listing to retrieve the correct cluster node pool models from the database.

I noticed I left a log line in the reivew fix force-with-lease push, so I fixed that here: https://github.com/banzaicloud/pipeline/compare/f640984ef5dee7ae5770cc55c9ccc9741c0e760b..9a918581fe55ba11f2ab3761d29d9f97eee75ed8.